src/core/core.cpp

```diff
@@ -373,12 +373,18 @@ uint32_t wf::compositor_core_impl_t::get_keyboard_modifiers()
 
 void wf::compositor_core_impl_t::set_cursor(std::string name)
```

Comment:
I'd prefer to put these checks in the wf::cursor_t functions - there are two of them there (one for clients, other for plugins, but both should have these guards).

Reply:
The checks can go in wf::cursor_t class and I did try this first but settled on this after having to move the counter out of wf::cursor_t class.